### PR TITLE
Use static linkage on Linux [ECR-3852]

### PR DESCRIPTION
**Pull request by vitvakatu**
## Overview

Use static linkage on both Linux and Mac, using the updated rocksdb library from Exonum Ubuntu PPA.

---
See: https://jira.bf.local/browse/ECR-3852

### Definition of Done

- [ ] There are no TODOs left in the code
- [ ] Change is covered by automated [tests](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#tests)
- [ ] The [coding guidelines](https://github.com/exonum/exonum-java-binding/blob/master/CONTRIBUTING.md#the-code-style) are followed
- [ ] Public API has Javadoc
- [ ] Method preconditions are checked and documented in the Javadoc of the method
- [ ] Changelog is updated if needed (in case of notable or breaking changes)
- [ ] The continuous integration build passes

### CODE REVIEW

File: exonum-java-binding/package_app.sh [hunk 1]
```diff
@@ -137,6 +123,23 @@ cp ./core/rust/exonum-java/log4j-fallback.xml "${PACKAGING_ETC_DIR}"
 # Copy tutorial
 cp ./core/rust/exonum-java/TUTORIAL.md "${PACKAGING_ETC_DIR}"
 
+# We use static linkage for RocksDB because in case of dynamic linking
+# the resulting app has a dependency on a _particular_
+# `install_name` of the RocksDB library. In case of RocksDB, `install_name`
```
Comment (dmitry-timofeev) [inline exonum-java-binding/package_app.sh]: What is the Linux alternative to `install_name` (I'd specify that once and go on using one of the terms)?

Comment (vitvakatu) [reply exonum-java-binding/package_app.sh]: It is called `SONAME`. I'd perhaps change to `SONAME` because Linux is definitely much more popular.

**Comment by vitvakatu** [comment]
> Do the contribution guide and installation instructions need to be updated?

No, using new `package_app` is enough. It is surprising that we do not have rocksdb library requirement for Linux. Perhaps it was a mistake, or we used static linkage with build-from-source rocksdb library.

**Comment by dmitry-timofeev** [comment]
Our "System Dependencies" section specifically mentions the RocksDB for Mac OS.
RocksDB also appears in Exonum installation instructions (which are linked in our instructions, but for Linux it is 5.17 from our PPA — will it work?

**Comment by vitvakatu** [comment]
> Our "System Dependencies" section specifically mentions the RocksDB for Mac OS.

Yes, because it is necessary for building EJB App from sources.

> RocksDB also appears in Exonum installation instructions (which are linked in our instructions, but for Linux it is 5.17 from our PPA — will it work?

It doesn't matter, the installed rocksdb version does not affect statically linked EJB App

**Comment by vitvakatu** [comment]
Updated Contributing.md as docs on exonum website seems outdated. Will send a PR to exonum-doc as well.

**Comment by dmitry-timofeev** [comment]
>
It doesn't matter, the installed rocksdb version does not affect statically linked EJB App

Then why were we 'blocked' by our PPA not having a static library?

**Comment by dmitry-timofeev** [comment]
@atrykush the application packaging job will likely need to be updated.

**Comment by vitvakatu** [comment]
> Then why were we 'blocked' by our PPA not having a static library?

Because, to package a statically linked EJB App, you have been required to:
1. Either download, compile and configure needed rocksdb version manually.
2. Or compile rocksdb from sources using internal rust-rocksdb library machinery, that takes a lot of time for each build.
3. Or use some other rocksdb version from some other PPA which has the required features enabled.

I'm not aware which of the options was used while packaging the App for Linux.

**Comment by dmitry-timofeev** [comment]
Then the installed rocksdb _does_ affect the packaging, for which we need proper instructions.

**Pull request merged**